source/MaterialXCore/Node.cpp

```diff
@@ -658,11 +664,28 @@ InputPtr Node::addInputFromNodeDef(const string& name)
         {
             nodeInput = addInput(nodeDefInput->getName());
             nodeInput->copyContentFrom(nodeDefInput);
+            for (auto nonInstanceAttribute : nonInstanceAttributes)
```

Comment:
It may be more straightforward here to provide a list of attributes to be _copied_ to instances, rather than copying all attributes and then removing the invalid ones afterwards.  This would be more robust as we add additional UI attributes in the future, not to mention situations where applications store custom attributes that we cannot know about.  What are your thoughts?

Reply:
I don't particularly like adding and then removing after the fact so the list of attributes to copy definitely is better -- and it could possibly make it so the cache is not dirtied as much.

For the "minimal" attributes that should be copied over from a nodedef to a node, I suggest:
- name
- type
- value